src/core/src/buffer.rs

```diff
@@ -103,6 +103,8 @@ pub enum Role {
     Index,
     /// Constant buffer
     Constant,
+    /// Staging buffer
+    Staging,
```

Comment:
I don't think it makes sense to be in `Role`. Staging is a property of underlying memory, not the role. Any role buffer can be stages.